src/toolkits/style_transfer/style_transfer_model_definition.cpp

```diff
@@ -0,0 +1,69 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#include <toolkits/style_transfer/style_transfer_model_definition.hpp>
+
+#include <ml/neural_net/weight_init.hpp>
+#include <toolkits/coreml_export/mlmodel_include.hpp>
+
+namespace turi {
+namespace style_transfer {
+
+using CoreML::Specification::InnerProductLayerParams;
+using CoreML::Specification::NeuralNetwork;
+using CoreML::Specification::NeuralNetworkLayer;
+
+using turi::neural_net::model_spec;
+using turi::neural_net::ones_weight_initializer;
+using turi::neural_net::zero_weight_initializer;
+
+namespace {
+
+std::unique_ptr<neural_net::model_spec> update_num_styles(
```

Comment:
Sigh, I see this is the first time we’re trying to modify parts of an existing model spec we’ve loaded from disk. (For OD, we just append to the pre-trained model.)

But I think we should avoid leaking raw protobuf code into this file. My proposal is this: instead programmatically construct the architecture of the model from scratch using the existing layer-by-layer `model_spec` methods. Initialize all the weights to zero, except for the instance norm weights, for which you do what you do here. Then use existing functionality to “merge” the parameters for the other layers from the pre-trained model into the model you built programmatically. 

“But Nick, why do go through all the trouble?” I think you’ll need to do this anyway to robustly support loading from a saved model. (Since our file format only saves the parameters by name, not the layer types.) So you may as well do that now, and avoid the leaky `model_spec` abstraction. Happy to discuss in more detail if you have other ideas or context, of course!

Reply:
This'll be fun! Haha!